pyproject.toml

```diff
@@ -1,9 +1,28 @@
 [tool.ruff]
 exclude = ["main_window.py", "home_ui.py", "resources.py", "resources_rc.py"]
+line-length = 100
+
+[tool.ruff.lint.per-file-ignores]
+"tagstudio/tests/**" = ["D"]
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
 
 [tool.ruff.lint]
-select = ["E", "F", "UP", "B", 'SIM']
-ignore = ["E402", "E501", "F541"]
+select = ["E", "F", "D", "I", "UP", "B", "SIM"]
+ignore = [
+	"E402",
+	"E501",
+	"F541",
```

Comment:
I originally put there this rule to avoid too many unrelated changes, but since we're going nuclear now, I'd enable this rule as well:

F541 - https://docs.astral.sh/ruff/rules/f-string-missing-placeholders/

Reply:
Ah yes, now's a good time to remove `F541`, I actually missed that it was still there.
`E402` shouldn't be a problem to remove, but I know there's some runtime imports for the v9.4 Blender thumbnailer... (problem for future me)
`E501` I agree should be removed, *but it doesn't mean I'm looking forward to the refactoring that it entails...*

Reply:
>  E501 I agree should be removed, but it doesn't mean I'm looking forward to the refactoring that it entails...

that should be handled automatically by `ruff format` anyway, no?

Reply:
Not in the cases of f-strings or long docstrings apparantly, which seem to make up most if not all of the cases being flagged. It's stuff that needs to happen, it's just a bit more involved.